src/iterators.jl

```diff
@@ -261,9 +356,77 @@ end
 @inline _getcache(fi::FaceIterator) = fi.fc
 @inline _getset(fi::FaceIterator) = fi.set
 
+"""
+    InterfaceIterator(grid::Grid, interfaces_set=1:length(topology.face_skeleton), topology::ExclusiveTopology)
+    InterfaceIterator(dh::AbstractDofHandler, interfaces_set=1:length(topology.face_skeleton), topology::ExclusiveTopology)
+Create an `InterfaceIterator` to conveniently iterate over all, or a subset, of the interfaces in a
+grid. The elements of the iterator are [`InterfaceCache`](@ref)s which are properly
+`reinit!`ialized. See [`InterfaceCache`](@ref) for more details.
+Looping over an `InterfaceIterator`, i.e.:
+```julia
+for ic in InterfaceIterator(grid, cellset)
+    # ...
+end
+```
+is thus simply convenience for the following equivalent snippet for grids of dimensions > 1:
+```julia
+ic = InterfaceCache(grid, topology)
+for face in topology.face_skeleton
+    neighborhood = topology.face_face_neighbor[face[1], face[2]]
+    isempty(neighborhood) && continue
+    neighbor_face = neighborhood[1]
+    reinit!(ic, face, neighbor_face)
+    # ...
+end
+```
+!!! warning
+    `InterfaceIterator` is stateful and should not be used for things other than `for`-looping
+    (e.g. broadcasting over, or collecting the iterator may yield unexpected results).
+"""
+struct InterfaceIterator{Cache<:InterfaceCache}
+    cache::Cache
+    set::Set{NTuple{2, FaceIndex}}
+end
+
+function InterfaceIterator(gridordh::Union{Grid,AbstractDofHandler},
```

Comment:
The set here is [Tuple([FaceIndex(cell_a, face_a), FaceIndex(cell_b, face_b)])...].
I think user can obtain it using faceset + topology.face_face_neighbor. and it's probably more consistent with how `FaceCache` uses `Set{FaceIndex}`. If you're ok with this approach I can document it.
Note that it does not check if the user passes "bad" faceindex (not neighbors?). the check happens when reiniting interfacevalues IIRC

Reply:
Perhaps the set functionality doesn't need to be here and we loop over all of them? You can always skip elements inside the loop:
```julia
for ic in InterfaceIterator(...)
    if some_condition(ic)
        continue
    end
    # do stuff
end
```

Reply:
Btw, should set be changed to a vector/Array as its size is already known and the elements are unique? It may have better performance IIUC